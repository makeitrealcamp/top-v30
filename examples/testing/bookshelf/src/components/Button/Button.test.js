import React from 'react'
import { render, screen, } from '@testing-library/react';
import { Button } from './'

describe('Button', () => {
  test('Should render correctly' , () => {
    render(<Button type='button'>Create</Button>)

    const buttonElement = screen.getByText('Create')

    expect(buttonElement).toBeInTheDocument()
    expect(buttonElement.type).toMatch('button')
  })

  test('Should match snapshot', () => {
    const { container } = render(<Button type='button'>Create</Button>)

    expect(container).toMatchSnapshot()
  })
})