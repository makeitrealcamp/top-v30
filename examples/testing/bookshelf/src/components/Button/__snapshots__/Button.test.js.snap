// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match snapshot 1`] = `
<div>
  <button
    data-testid="button"
    type="button"
  >
    Create
  </button>
</div>
`;
